src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,143 @@
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  // Reference to Firestore database
+  private Firestore firestoreDb;
+
+  // Reference to Realtime database
+  private DatabaseReference realtimeDb;
+
+  // Counter that will be updated 
+  Counter counter = new Counter(0);
+
+  @Override
+  public void init() {
+
+    try {
+      // Initialization of reference to Realtime database
+
+      // Building options, include credentials and which DB to use
+      FirebaseOptions options = new FirebaseOptions.Builder()
+          .setCredentials(GoogleCredentials.getApplicationDefault())
+          .setDatabaseUrl("https://quizzy-step-2020.firebaseio.com/")
+          .build();
+
+      // Initialize DB with options
+      FirebaseApp.initializeApp(options);
+
+      // Reference to the whole Realtime DB
+      final FirebaseDatabase database = FirebaseDatabase.getInstance();
+      
+      // Reference to the "counter" object
+      realtimeDb = database.getReference("/users-counter/counter");
+
+      // Add a listener to the counter, executed when counter is changed
+      realtimeDb.addValueEventListener(new ValueEventListener() {
+        @Override
+        public void onDataChange(DataSnapshot dataSnapshot) {
+          counter = dataSnapshot.getValue(Counter.class);
+          System.out.println("Listener executed " + counter.value);
+        }
+      
+        @Override
+        public void onCancelled(DatabaseError databaseError) {
+          System.out.println("The read failed: " + databaseError.getCode());
+        }
+      });
+
+
+      // Initialization of Firestore database
+
+      // Building options, include projectID, access credentials
+      FirestoreOptions firestoreOptions =
+      FirestoreOptions.getDefaultInstance().toBuilder()
+          .setProjectId("quizzy-step-2020")
+          .setCredentials(GoogleCredentials.getApplicationDefault())
+          .build();
+
+      // Get database and store the reference in firestoreDb variable
+      firestoreDb = firestoreOptions.getService();
+
+    } catch(IOException ie) {
+      ie.printStackTrace();
+    }
+
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Writting to the DBs</h1>");
+
+    // Writing to Realtime DB
+
+    // Creating a tempCounter just to show that counter is not updated locally, but in the DB
+    Counter tempCounter = new Counter(counter.value + 1);
+    response.getWriter().println(tempCounter.value);
+
+    // Update counter asynchronously in Realtime DB with no callback
+    realtimeDb.setValueAsync(tempCounter);
+
+    // Set value and attach a callback for when it is ready, may not work because of AppEngine thread management?
```

Comment:
So does it seem like both of these work or just one?

Reply:
Ohhh man I forgot to comment that out, both work!

Reply:
Done.

Reply:
Fix comment?

Reply:
Uhh my last commit commented out that line, why isn't it showing that?

Reply:
Nah I just mean remove/correct "may not work because of AppEngine thread management?"